ert_shared/ensemble_evaluator/config.py

```diff
@@ -26,34 +27,37 @@ def find_open_port(lower, upper):
     raise Exception(msg)
 
 
-CLIENT_URI = "client"
-DISPATCH_URI = "dispatch"
-DEFAULT_PORT = find_open_port(lower=51820, upper=51840)
-DEFAULT_HOST = _get_ip_address()
-DEFAULT_URL = f"ws://{DEFAULT_HOST}:{DEFAULT_PORT}"
-CONFIG_FILE = "ee_config.yml"
-
-DEFAULT_EE_CONFIG = {
-    "host": DEFAULT_HOST,
-    "port": DEFAULT_PORT,
-    "url": DEFAULT_URL,
-    "client_url": f"{DEFAULT_URL}/{CLIENT_URI}",
-    "dispatch_url": f"{DEFAULT_URL}/{DISPATCH_URI}",
-}
-
-
-def load_config(config_path=None):
-    if config_path is None:
-        return DEFAULT_EE_CONFIG.copy()
-
-    with open(config_path, "r") as f:
-        data = yaml.safe_load(f)
-        host = data.get("host", DEFAULT_HOST)
-        port = data.get("port", DEFAULT_PORT)
-        return {
-            "host": host,
-            "port": port,
-            "url": f"ws://{host}:{port}",
-            "client_url": f"ws://{host}:{port}/{CLIENT_URI}",
-            "dispatch_url": f"ws://{host}:{port}/{DISPATCH_URI}",
-        }
+class EvaluatorServerConfig:
```

Comment:
Do we want this class to behave as a singleton? If the load_config function is called one place without the config path, and then another place with the config path, then the second call will potentially overwrite class properties. This could lead to very tricky bugs to find. Maybe we could return an instance with the specific configurations loaded, and then pass that where we need it (or even better just the parts that are needed)

Reply:
```
If the load_config function is called one place without the config path, and then another place with the config path, then the second call will potentially overwrite class properties.
```

This is true, class properties will also be overwritten if you load from different files throughout the code,  it seems a bit poorly designed to allow both loading a config from a file and also without a file providing default values given the issue with reserving the port. 

```
Maybe we could return an instance with the specific configurations loaded, and then pass that where we need it (or even better just the parts that are needed) 
```
Very interesting idea, I like it I will experiment with it.